src/FSharp.AWS.DynamoDB/TableContext.fs

```diff
@@ -343,11 +332,10 @@ type TableContext<'TRecord> internal
         let writeRequests = items |> Seq.map mkWriteRequest |> rlist
         let pbr = BatchWriteItemRequest()
         pbr.RequestItems.[tableName] <- writeRequests
-        let returnConsumedCapacity, maybeReport = metricsOptions collector
         pbr.ReturnConsumedCapacity <- returnConsumedCapacity
         let! ct = Async.CancellationToken
         let! response = client.BatchWriteItemAsync(pbr, ct) |> Async.AwaitTaskCorrect
-        match maybeReport with None -> () | Some r -> r BatchWriteItems (Seq.toList response.ConsumedCapacity) 0
+        maybeReport |> Option.iter (fun r -> r BatchWriteItems (Seq.toList response.ConsumedCapacity) (items.Length - response.UnprocessedItems.Count))
```

Comment:
Good catch, thanks

Reply:
Hm, if `UnprocessedItems` is a dict of table -> seq key,  then this will always incorrectly return 1 ?

Reply:
It’s not _always_ incorrect, only if there’s more than one unprocessed 😉 - nice catch, thanks. Unfortunately I don’t think there’s a way to test unprocessed using local; it’s intended for throughput limits.

Reply:
Would be in favour of inlining the body of `unprocessedPutAttributeValues`  so it's not off in another file :-
```
        let unprocessed =
            match response.UnprocessedItems.TryGetValue tableName with
            | true, reqs -> reqs |> Seq.choose (fun r -> Option.ofObj r.PutRequest) |> Seq.toArray
            | false, _ -> Array.empty
 ```
 then you can just use `unprocessed.Length` ?